tools/p4mlir-translate/translate.cpp

```diff
@@ -1713,6 +1742,42 @@ bool P4HIRConverter::preorder(const P4::IR::AssignmentStatement *assign) {
     return false;
 }
 
+bool P4HIRConverter::expandOpAssignBinOp(const P4::IR::OpAssignmentStatement *opAssign,
+                                         P4HIR::BinOpKind kind) {
+    auto loc = getLoc(builder, opAssign);
+    auto lhsRef = resolveReference(opAssign->left);
+    visit(opAssign->right);
+
+    auto binop = builder.create<P4HIR::BinOp>(
+        loc, kind, getValue(opAssign->left, getOrCreateType(opAssign->left->type)),
+        getValue(opAssign->right, getOrCreateType(opAssign->right->type)));
```

Comment:
You'd likely need to insert cast here, but for left type here. Add tests where rhs is serialized enum.